source/frontend/src/components/common/mapFSM/machineDefinition/types.ts

```diff
@@ -0,0 +1,59 @@
+import { LatLngBounds, LatLngLiteral } from 'leaflet';
+
+import { IPropertyFilter } from '@/features/properties/filter/IPropertyFilter';
+
+import { FeatureSelected, MapFeatureData, RequestedFlyTo } from '../models';
+import { LocationFeatureDataset } from '../useLoactionFeatureLoader';
+
+export enum SideBarType {
+  NOT_DEFINED = 'NOT_DEFINED',
+  MAP = 'MAP',
+  RESEARCH_FILE = 'RESEARCH_FILE',
+  ACQUISITION_FILE = 'ACQUISITION_FILE',
+  LEASE_FILE = 'LEASE_FILE',
+  PROJECT = 'PROJECT',
+  PROPERTY_INFORMATION = 'PROPERTY_INFORMATION',
+}
+
+// Local context for the machine - Not related to React Context!
+export type MachineContext = {
+  isSelecting: boolean;
+  mapFeatureSelected: FeatureSelected | null;
+  mapLocationSelected: LatLngLiteral | null;
+  mapLocationFeatureDataset: LocationFeatureDataset | null;
+  selectedFeatureDataset: LocationFeatureDataset | null;
+
+  mapFeatureData: MapFeatureData;
+
+  // TODO: this is partially in the URL. Either move it completly there or remove it
+  mapFilter: IPropertyFilter | null;
+
+  isLoading: boolean;
+  sideBarType: SideBarType;
+  requestedFitBounds: LatLngBounds;
+  requestedFlyTo: RequestedFlyTo;
+  draftLocations: LatLngLiteral[];
+};
+
+// Possible state machine states
```

Comment:
background on this?

Reply:
This will be needed when the state machine becomes strictly typed.